src/contracts/middleware/BLSIndexRegistryCoordinator.sol

```diff
@@ -0,0 +1,163 @@
+     * @param globalOperatorListIndex is the operator's index in the global operator list in the IndexRegistry
+     */
+    function deregisterOperator(BN254.G1Point memory pubkey, uint32[] memory quorumToOperatorListIndexes, uint32 globalOperatorListIndex) external {
+        _deregisterOperator(msg.sender, pubkey, quorumToOperatorListIndexes, globalOperatorListIndex);
+    }
+
+    function _registerOperator(address operator, bytes calldata quorumNumbers, BN254.G1Point memory pubkey) internal {
+        // TODO: check that the sender is not already registered
+
+        // get the quorum bitmap from the quorum numbers
+        uint256 quorumBitmap = BytesArrayBitmaps.orderedBytesArrayToBitmap_Yul(quorumNumbers);
+        require(quorumBitmap != 0, "BLSIndexRegistryCoordinator: quorumBitmap cannot be 0");
+
+        // register the operator with the BLSPubkeyRegistry and get the operatorId (in this case, the pubkeyHash) back
+        bytes32 operatorId = blsPubkeyRegistry.registerOperator(operator, quorumNumbers, pubkey);
+
+        // register the operator with the IndexRegistry
+        indexRegistry.registerOperator(operatorId, quorumNumbers);
+
+        // register the operator with the StakeRegistry
+        _registerOperator(operator, operatorId, quorumNumbers);
+
+        // set the operatorId to quorum bitmap mapping
+        operatorIdToQuorumBitmap[operatorId] = quorumBitmap;
+
+        // set the operator struct
+        operators[operator] = Operator({
+            operatorId: operatorId,
+            fromTaskNumber: serviceManager.taskNumber(),
+            status: OperatorStatus.REGISTERED
+        });
+    }
+
+    function _deregisterOperator(address operator, BN254.G1Point memory pubkey, uint32[] memory quorumToOperatorListIndexes, uint32 globalOperatorListIndex) internal {
+        require(operators[operator].status == OperatorStatus.REGISTERED, "BLSIndexRegistryCoordinator._deregisterOperator: operator is not registered");
+
+        // get the operatorId of the operator
+        bytes32 operatorId = operators[operator].operatorId;
+        require(operatorId == pubkey.hashG1Point(), "BLSIndexRegistryCoordinator._deregisterOperator: operatorId does not match pubkey hash");
+
+        // get the quorumNumbers of the operator
+        bytes memory quorumNumbers = BytesArrayBitmaps.bitmapToBytesArray(operatorIdToQuorumBitmap[operatorId]);
+        
+        // deregister the operator from the BLSPubkeyRegistry
+        blsPubkeyRegistry.deregisterOperator(operator, quorumNumbers, pubkey);
+
+        // deregister the operator from the IndexRegistry
+        indexRegistry.deregisterOperator(operatorId, quorumNumbers, quorumToOperatorListIndexes, globalOperatorListIndex);
+
+        // deregister the operator from the StakeRegistry
+        _deregisterOperator(operator, operatorId, quorumNumbers);
```

Comment:
see my comment above about the 2 functions having the same name. Similarly, having 2 `_deregisterOperator` functions feels like an anti-pattern that we should fix.

Reply:
totally, didn't even realize this. here's my proposed fix: https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/0c907f468e08792b5d0db933ba1a12c64c4c9d32